source/configuration/modules/imfile.rst

```diff
@@ -1,570 +1,408 @@
-   a different value will be ignored, but will generate an error
-   message.
-
-   Please note that future versions of imfile may not support this
-   parameter at all. So it is suggested to not use it.
-
-   In *polling* mode, if set to 0, each file will be fully processed and
-   then processing switches to the next file. If it is set to any other
-   value, a maximum of [number] lines is processed in sequence for each file,
-   and then the file is switched. This provides a kind of mutiplexing
-   the load of multiple files and probably leads to a more natural
-   distribution of events when multiple busy files are monitored. For
-   *polling* mode, the **default** is 10240.
-
-.. index:: 
-   single: imfile; MaxSubmitAtOnce
-.. function:: MaxSubmitAtOnce [number]
-
-   This is an expert option. It can be used to set the maximum input
-   batch size that imfile can generate. The **default** is 1024, which
-   is suitable for a wide range of applications. Be sure to understand
-   rsyslog message batch processing before you modify this option. If
-   you do not know what this doc here talks about, this is a good
-   indication that you should NOT modify the default.
+*****************
 
-.. index:: 
-   single: imfile;  deleteStateOnFileDelete
-.. function:: deleteStateOnFileDelete [on/off] (requires v8.5.0+)
-
-   **Default: on**
-
-   This parameter controls if state files are deleted if their associated
-   main file is deleted. Usually, this is a good idea, because otherwise
-   problems would occur if a new file with the same name is created. In
-   that case, imfile would pick up reading from the last position in
-   the **deleted** file, which usually is not what you want.
-
-   However, there is one situation where not deleting associated state
-   file makes sense: this is the case if a monitored file is modified
-   with an editor (like vi or gedit). Most editors write out modifications
-   by deleting the old file and creating a new now. If the state file
-   would be deleted in that case, all of the file would be reprocessed,
-   something that's probably not intended in most case. As a side-note,
-   it is strongly suggested *not* to modify monitored files with
-   editors. In any case, in such a situation, it makes sense to
-   disable state file deletion. That also applies to similar use
-   cases.
+================  ==================
+mode:             "inotify|polling"
+Default:          "inotify"
+Legacy:           $InputFileReadMode
+Available since:  8.1.5
+================  ==================
 
-   In general, this parameter should only by set if the users
-   knows exactly why this is required.
+This specifies if imfile is shall run in *inotify* or *polling* mode.
 
-.. index:: 
-   single: imfile;  Ruleset
-.. function:: Ruleset <ruleset> 
+inotify:
 
-   Binds the listener to a specific :doc:`ruleset <../../concepts/multi_ruleset>`.
+**TODO add a description about how inotify works internally**
```

Comment:
Polling works by waking up every X seconds and checking for new data in the file. Inotify works by registering with the OS to be notified when the file changes.

Inotify is less load on the system than polling, but it does require more support from the kernel and filesystem, so there can be cases where polling is desired